### ISSUE TITLE
Capitalize does not follow Shopify spec - capitalizes each word

### ISSUE THREAD

**Issue by daviburg** [dotliquid/dotliquid#390]
### Dotliquid version
Current as of 6-13-2020

### Expected behavior
{{ "my great title" | capitalize }} outputs My great title
as per https://shopify.github.io/liquid/filters/capitalize/

### Actual behavior
{{ "my great title" | capitalize }} outputs My Great Title

### Steps to reproduce the Problem (you can add files)
            // Following test disabled due to out-of-spec bug
            //// Helper.AssertTemplateResult(
            ////     expected: "My great title",
            ////     template: "{{ 'my great title' | capitalize }}");

**Comment by microalps** [comment]
CSS has a text transform named identically that behaves like DotLiquid. The spec is out of line with what users expect. Maybe we should add CapitalizeFirst which is explicit and in line with other filters like ReplaceFirst

**Comment by daviburg** [comment]
Agree we need an approach that is non-breaking.
At the same time to facilitate re-use of Liquid templates from one implementation to the other, we need to have one implementation that is consistent with Shopify's (maybe with a flag 'StrictShopify').

**Comment by microalps** [comment]
Does PR #420 fully resolve this issue?

**Comment by daviburg** [comment]
> Does PR #420 fully resolve this issue?

yes

**Comment by microalps** [comment]
Shopify is working on updating their docs (see https://github.com/Shopify/liquid/pull/1403 and the future docs at https://ericfromcanada.github.io/liquid/filters/capitalize/) that notes that capitalize actually lower cases everything. So mcDonald becomes Mcdonald! LiquidJS does not implement this way and is aligned with how we implemented in 2.1. However, as this does not match, this issue is being reopened.

I tested Ruby liquid on my server - 
```
{
  "template": "{{ name | capitalize }}",
  "data": {
    "name": "John G. Chalmers-Smith"
  }
}
```
Result: _John g. chalmers-smith_
if I add a space before the name, it will be all lowercase.
